### PR TITLE
Pytest: default settings on PC improvements

### PR DESCRIPTION
**Pull request by jnewb1**
- xdist by default on pc
- run cpp tests by default on pc
- separate config for TICI, so we don't have to wait for an agnos release before adding pytest features

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Can you make sure agnos-builder is up to date with our pytest packages?

File: system/sensord/tests/test_sensord.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def setUpClass(cls):
     os.environ["LSM_SELF_TEST"] = "1"
 
     # read initial sensor values every test case can use
-    os.system("pkill -f ./sensord")
+    os.system("pkill -f \\.\\/sensord")
```
Comment (jnewb1) [inline system/sensord/tests/test_sensord.py:109]: this was killing pytest since its a regex

**Comment by sshane** [comment]
This broke multithreading pytest on local PC.

**Comment by jnewb1** [comment]
> This broke multithreaded pytest on local PC.
> 
> `pytest -n 32 ../car/tests/test_car_interfaces.py ` is single-threaded for example and takes a minute where it previously took seconds. This is from `--dist=loadscope` that is only really required for test_models since its `setUpClass` is extremely slow.

hmm, you should still be able to override it with --dist=load, but perhaps load should be the default?
`pytest -n 32 --dist=load  ../car/tests/test_car_interfaces.py `

**Pull request merged**